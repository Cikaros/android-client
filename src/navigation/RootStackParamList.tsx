export type RootStackParamList = {
  Home: undefined;
  Onboarding: undefined;
  ChangeServer: undefined;
  LeftDrawer: undefined;
  About: undefined,
  Advance: undefined,
  Peers: undefined,
};
