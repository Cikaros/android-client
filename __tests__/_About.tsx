import React from 'react';
import {shallow} from 'enzyme';
import {Provider} from 'react-redux';
import About from '../src/screens/About/index';
import {RootStackParamList} from '@navigation/RootStackParamList';
import {StackNavigationProp} from '@react-navigation/stack';
import {RouteProp} from '@react-navigation/native';
import configureStore from 'redux-mock-store';

const mockStore = configureStore();
const initialState = {}; 
const store = mockStore(initialState);

const navigation = {
  navigate: jest.fn(),
  goBack: jest.fn(),
  dispatch: jest.fn(),
  reset: jest.fn(),
  isFocused: jest.fn(),
  canGoBack: jest.fn(),
} as unknown as StackNavigationProp<RootStackParamList, 'About'>;

const route: RouteProp<RootStackParamList, 'About'> = {
  key: 'mock-key',
  name: 'About',
};

// Mock the Linking.openURL method
jest.mock('react-native', () => ({
  ...jest.requireActual('react-native'),
  Linking: {
    openURL: jest.fn(),
  },
}));

jest.mock('react-native-device-info', () => ({
  getVersion: jest.fn(() => '1.0.1'), // Mock the getVersion function to return a specific value
}));

describe('About Screen', () => {
  it('displays the correct version', () => {
    const wrapper = shallow(
      <Provider store={store}>
        <About navigation={navigation} route={route} />
      </Provider>,
    );
    const textComponents = wrapper.find('Text');

    expect(textComponents.at(0).contains('Version'));
    const versionText = wrapper.findWhere(
      node => node.prop('testID') === 'versionText',
    );
    expect(versionText).toBeDefined();
    expect(textComponents.at(0).contains('1.0.1'));
    expect(textComponents.at(1).contains('License agreement'));
    expect(textComponents.at(2).contains('Privacy policy'));
    expect(textComponents.at(3).contains('Â© 2023 NetBird all rights reserved'));
    expect(wrapper).toMatchSnapshot();
  });

});
