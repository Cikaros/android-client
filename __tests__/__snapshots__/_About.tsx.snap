// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About Screen displays the correct version 1`] = `
<ContextProvider
  value={
    {
      "getServerState": undefined,
      "noopCheck": "once",
      "stabilityCheck": "once",
      "store": {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "subscription": {
        "addNestedSub": [Function],
        "getListeners": [Function],
        "handleChangeWrapper": [Function],
        "isSubscribed": [Function],
        "notifyNestedSubs": [Function],
        "trySubscribe": [Function],
        "tryUnsubscribe": [Function],
      },
    }
  }
>
  <About
    navigation={
      {
        "canGoBack": [MockFunction],
        "dispatch": [MockFunction],
        "goBack": [MockFunction],
        "isFocused": [MockFunction],
        "navigate": [MockFunction],
        "reset": [MockFunction],
      }
    }
    route={
      {
        "key": "mock-key",
        "name": "About",
      }
    }
  />
</ContextProvider>
`;
